WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.h

```diff
@@ -1,13 +1,31 @@
 #import <UIKit/UIKit.h>
 
+
+typedef void (^SettingsTextChanged)(NSString *);
+
+// Reusable component that renders a UITextField + Hint onscreen. Useful for Text / Password / Email data entry.
+//
 @interface SettingsTextViewController : UITableViewController
 
-@property (nonatomic, copy) void(^onValueChanged)(NSString *);
-@property (nonatomic, copy) void(^onCancel)();
+// Block to be executed on dismiss, if the value was effectively updated.
+//
+@property (nonatomic, copy) SettingsTextChanged onValueChanged;
+
+// Property to indicate whether email validation should be performed, or not.
+//
+@property (nonatomic, assign) BOOL isEmail;
```

Comment:
I was already on the fence with `isPassword` but I think this controller is starting to do too much. What happens when you also want to edit an URL? What happens if you set bot `isEmail` and `isPassword` to `YES`?

I think the original design of showing a text field with an optional placeholder/help text, and offering a callback on dismiss was spot on. Then we've added specific cases complicating it too much. If you think about it, what's really specific about password/email/URL is:

1. Traits of the text field: not just `secureTextEntry`, but email/URL should set the appropriate keyboard.
2. Validation: is this a valid email address/URL? is it a long enough password?

I'd propose to extract all of that to a separate class.